spec/services/work_packages/set_schedule_service_working_days_spec.rb

```diff
@@ -0,0 +1,1200 @@
+            work_package  | X          |
+          CHART
+        end
+
+        it 'reschedules the follower to start after the non-working days' do
+          expect_schedule(subject.all_results, <<~CHART)
+                          |    MTWTFSS |
+            work_package  | X          |
+            follower      |    XX      |
+          CHART
+        end
+      end
+
+      context 'on a day in the middle of working days' do
+        let_schedule(<<~CHART, ignore_non_working_days: false)
+          days          | MTWTFSS |
+          work_package  | X       |
+          follower      |  XX     | follows work_package
+        CHART
+
+        before do
+          change_schedule([work_package], <<~CHART)
+            days          | mtwtfssmtwtfssMTWTFSS |
+            work_package  |  X                    |
+          CHART
+        end
+
+        it 'reschedules the follower to move by the same delta of working days' do
+          expect_schedule(subject.all_results, <<~CHART)
+            days          | mtwtfssmtwtfssMTWTFSS |
+            work_package  |  X                    |
+            follower      |   XX                  |
+          CHART
+        end
+      end
+
+      context 'on a day before non-working days the follower having space between' do
+        let_schedule(<<~CHART, ignore_non_working_days: false)
+          days          | MTWTFSS   |
+          work_package  | X         |
+          follower      |     X     | follows work_package
+        CHART
+
+        before do
+          change_schedule([work_package], <<~CHART)
+            days          |    MTWTFSS |
+            work_package  | X          |
+          CHART
+        end
+
+        it 'reschedules follower to move backward by the same delta of working days' do
+          expect_schedule(subject.all_results, <<~CHART)
+                          |    MTWTFSS   |
+            work_package  | X            |
+            follower      |       X      |
+          CHART
+        end
+      end
+
+      context 'with the follower having another relation limiting movement' do
+        let_schedule(<<~CHART, ignore_non_working_days: false)
+          days          | mtwtfssmtwtfssMTWTFSS |
+          work_package  |               X       |
+          follower      |                XX     | follows work_package, follows annoyer with delay 2
+          annoyer       |    XX..XX             |
+        CHART
+
+        before do
+          change_schedule([work_package], <<~CHART)
+            days          | mtwtfssmtwtfssMTWTFSS |
+            work_package  |  X            _       |
+          CHART
+        end
+
+        it 'reschedules follower to move backward but not earlier than the other relation soonest start date' do
+          expect_schedule(subject.all_results, <<~CHART)
+            days          | mtwtfssmtwtfssMTWTFSS |
+            work_package  |  X                    |
+            follower      |            X..X       |
```

Comment:
```suggestion
            follower      |            X..X       |
            annoyer       |    XX..XX             |
```
Since we are also testing that the follower has a 2 days distance from the `annoyer`, I would add the `annoyer` to the expected chart for better visual cueing.

Reply:
If we do that, then the test will fail because `annoyer` was not present in `subject.all_results`. I agree it would be better to have it for better visual cueing, but I did not find a good way to do so. Ideas? 

My current idea: maybe a `#` before the line to ignore it? But then it would not be tested. What if the line gets out of date?